File: inventario_frontend/src/models/product.ts
import { Category } from "./category";
import { Vendor } from "./vendor";
import { PriceForCreation, Price } from "./price";

export class ProductForCreation {
  public name: string | undefined = undefined;
  public description: string | undefined = undefined;
  public code?: string;
  public brand: string | undefined = undefined;
  public categoryId: string | undefined = undefined;
  public category?: Category;
  public vendorId?: string;
  public vendor?: Vendor;
  public purchasePrice: PriceForCreation;
  public salePrice: PriceForCreation;
  public minimumStock?: number;
  public stock = 0;
  public unitOfMeasurement: string | undefined = undefined;

  constructor() {
    this.purchasePrice = new PriceForCreation();
    this.salePrice = new PriceForCreation();
    this.stock = 0;
  }
}

export class Product extends ProductForCreation {
  public id!: string;
  public purchasePrice: Price;
  public salePrice: Price;

  constructor() {
    super();
    this.purchasePrice = new Price();
    this.salePrice = new Price();
  }
}

File: Inventario/DataAccess/Models/Product.cs
using Microsoft.VisualBasic;
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.ComponentModel.DataAnnotations;
using System.Text;
using Util.Enums;

namespace Data
{
    public class Product: Entity
    {
        [Required]
        public string Name { get; set; }
        [Required]
        public string Description { get; set; }
        public string Code { get; set; }
        [Required]
        public Guid CategoryId { get; set; }
        public Category Category { get; set; }
        public Guid VendorId { get; set; }
        public Vendor Vendor { get; set; }
        public string Brand { get; set; }
        public double Stock { get; set; }
        public string UnitOfMeasurement { get; set; }
    }
}
